File: src/api/LMSEntities/DataTransferObjects/LibraryCardForCreationDto.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace LMSEntities.DataTransferObjects
{
    public class LibraryCardForCreationDto
    {
        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string Email { get; set; }

        [Phone]
        public string PhoneNumber { get; set; }

        public DateTime DateOfBirth { get; set; }

        [Required]
        public AddressDto Address { get; set; }
        [Required]
        public MemberGenderDto Gender { get; set; }
    }
}

File: src/spa/src/app/dto/models/library-card-for-creation-dto.ts
/* tslint:disable */
/* eslint-disable */
import { AddressDto } from './address-dto';
import { MemberGenderDto } from './member-gender-dto';
export interface LibraryCardForCreationDto {
  address: AddressDto;
  dateOfBirth?: string;
  email?: null | string;
  firstName?: null | string;
  gender: MemberGenderDto;
  lastName?: null | string;
  phoneNumber?: null | string;
}
